apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -477,7 +500,11 @@ export default NextAuth({
         if (existingUserWithEmail) {
           // if self-hosted then we can allow auto-merge of identity providers if email is verified
           if (!hostedCal && existingUserWithEmail.emailVerified) {
-            return true;
+            if (existingUserWithEmail.twoFactorEnabled) {
+              return `/auth/login?totp=${await signJwt({ email: existingUserWithEmail.email })}`;
```

Comment:
This is the redirection needed to let the user input a totp code in an external provider use case.